sqlmesh/dbt/adapter.py

```diff
@@ -98,14 +98,31 @@ def quote(self, identifier: str) -> str:
     def dispatch(self, name: str, package: t.Optional[str] = None) -> t.Callable:
         """Returns a dialect-specific version of a macro with the given name."""
         target_type = self.jinja_globals["target"]["type"]
-        references_to_try = [
-            MacroReference(package=f"{package}_{target_type}", name=f"{target_type}__{name}"),
-            MacroReference(package=package, name=f"{target_type}__{name}"),
-            MacroReference(package=package, name=f"default__{name}"),
+        macro_suffix = f"__{name}"
+
+        def _macro_relevance(val: t.Tuple[str, str]) -> int:
+            """Lower scores more relevant."""
+            package_name, macro_name = val
+            score = 0
+            if package_name != package:
+                score += 2
```

Comment:
I think this mechanics is overly complex. Can we just stick to 3 states: 0-2 with 2 being `default__`, 1 for the same package name and `__{name}`, and 0 for `{package}__` + `__{name}`

Reply:
Sorry, I meant in reversed order.

Reply:
The above suggestion is missing different package + expected name. For example, `dbt_redshift.redshift__{name}` would be the correct macro and should be preferred over `dbt.postgres__{name}`.

Reply:
I ranked the best match as 0, so I didn't have to add reverse to the sort. Do you have a strong opinion here?

Reply:
> I ranked the best match as 0, so I didn't have to add reverse to the sort. Do you have a strong opinion here?

No, I don't.

> For example, dbt_redshift.redshift__{name} would be the correct macro and should be preferred over dbt.postgres__{name}.

Doesn't this:

> and 0 for `{package}__` + `__{name}`

cover it?